### DISCUSSION TITLE
General program for converting attitude data to ORBEX

### DISCUSSION THREAD

**Discussion by AndreHauschild** [groops-devs/groops#256] [Ideas]
Dear Groops-Team,

the program `GnssAttitude2Orbex` is intended to convert attitude instrument files of GNSS satellites into ORBEX format. It would be helpful if LEO satellite attitude data could also be converted into ORBEX. For this a more generalized program `StarCamera2Orbex` with a similar interface as `Orbit2Sp3Format` could be used. I would appreciate if you could consider adding this to the groops programs. Many thanks!

**Comment by AndreHauschild** [comment]
Dear groops community,

for a more flexible conversion of star camera files into ORBEX format I have created a new program `starCamera2Orbex`, which handles satellite identifiers in the same way as `Orbit2Sp3Format`. This way, the input star camera files may contain a satellite name (or any other identifier) and the corresponding PRN is then used in the resulting ORBEX file. The current conversion program `GnssAttitude2Orbex` does not offer this feature, so `starCamera2Orbex` could potentially supersede it. The source code can be found in [this branch](https://github.com/AndreHauschild/groops/tree/starCamera2Orbex).

I have tested the new program by converting a number of star camera files to ORBEX, then converted back to star camera files and compared the input and output in terms of Euler angles differences. The maximum differences are always less than 1e-11 deg. The test script is attached. 
[010_testStarCamera2Orbex.xml.gz](https://github.com/user-attachments/files/17494286/010_testStarCamera2Orbex.xml.gz)

I would very much appreciate your comments and feedback!

**Comment by tmayerguerr** [comment]
Dear Andre,
Many thanks for your contribution. Its now merged into main. StarCamera2Orbex replaces GnssAttitude2Orbex, which is now deprecated.
best regards
Torsten
